code/game/objects/random/random.dm

```diff
@@ -53,3 +53,13 @@
 
 /obj/random/single/item_to_spawn()
 	return ispath(spawn_object) ? spawn_object : text2path(spawn_object)
+
+/obj/randomcatcher
```

Comment:
What is this catcher thing all about?

Reply:
Sometimes random types spawn a random type which then spawns an item. The catcher is used in these cases to get the proper item instead of getting the result of the spawn which would be a random type(which is intended to yet spawn another item)